src/commonMain/kotlin/au/id/micolous/metrodroid/transit/serialonly/HoloTransitData.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * HoloTransitData.kt
+ *
+ * Copyright 2015-2016 Michael Farrell <micolous+git@gmail.com>
+ * Copyright 2018 Google Inc.
+ *
+ * Authors: Vladimir Serbinenko, Michael Farrell, Trevor Nielsen
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package au.id.micolous.metrodroid.transit.serialonly
+
+import au.id.micolous.metrodroid.card.CardType
+import au.id.micolous.metrodroid.card.desfire.DesfireApplication
+import au.id.micolous.metrodroid.card.desfire.DesfireCard
+import au.id.micolous.metrodroid.card.desfire.DesfireCardTransitFactory
+import au.id.micolous.metrodroid.multi.Parcelize
+import au.id.micolous.metrodroid.multi.R
+import au.id.micolous.metrodroid.time.Epoch
+import au.id.micolous.metrodroid.time.MetroTimeZone
+import au.id.micolous.metrodroid.time.TimestampFormatter
+import au.id.micolous.metrodroid.transit.CardInfo
+import au.id.micolous.metrodroid.transit.TransitIdentity
+import au.id.micolous.metrodroid.transit.TransitRegion
+import au.id.micolous.metrodroid.ui.ListItem
+
+/**
+ * Transit data type for HOLO card.
+ *
+ *
+ * This is a very limited implementation of reading HOLO, because only
+ * little data is stored on the card
+ *
+ *
+ * Documentation of format: https://github.com/metrodroid/metrodroid/wiki/HOLO
+ */
+@Parcelize
+data class HoloTransitData(private val mSerial: Int?,
+                                private val mLastTransactionTimestamp: Int?, private val mManufacturingId: String?) : SerialOnlyTransitData() {
+
+    public override val extraInfo: List<ListItem>
+        get() = listOf(
+            ListItem(R.string.last_transaction, when (mLastTransactionTimestamp) {
+                0 -> "Never"
+                null -> "Never"
+                else -> TimestampFormatter.dateTimeFormat(parseTime(mLastTransactionTimestamp)).toString()
+            }),
+            ListItem(R.string.manufacture_id, mManufacturingId?.let { formatMfgId(mManufacturingId)})
+            )
+
+    override val reason: Reason
+        get() = Reason.NOT_STORED

```

Comment:
Did you check that
a) no files change with changed balance
b) There are no locked files
Can you send me couple of annotated dumps (e.g. before and after transaction)?

Reply:
a) Based on the dumps that @frozenpandaman annotated and a dump from a card I have there seems to be no balance stored.
b) There are no files with encrypted read.
Here's a full dump from a Proxmark of a card that hasn't been used yet.
```
[usb] pm3 --> hf mfdes dump --aid f21360 --no-auth -v
[=] Key num: 0 Key algo: des Key[8]: 00 00 00 00 00 00 00 00
[=] Secure channel: n/a Command set: niso Communication mode: plain
[=] AID f21360 is selected
[!!] Desfire GetFileISOIDList command error. Result: -20
[!] ISO ID list returned no data

[+] Application AID f21360 have 2 files
[+]   AID mapped to MIFARE Classic AID (MAD): 2136
[+]   MAD AID Cluster  0x21      : reserved
[=]   MAD AID Function 0x2136    : (unknown)

[=] --------------------------------- File 00 ----------------------------------
[+] File ID         : 00
[+] File type       : Standard data  [0x00]
[+] Comm mode       : Plain
[+] File size       : 96 [0x60] bytes
[+] Access rights   : eff0  (r: free w: disabled rw: disabled change: key 0x00)
[=] Got file type: Standard data. Option: data. comm mode: plain
[=] ------------------------------- File 00 data -------------------------------
[+] Read 96 bytes from file 0x00 offset 0
[=]  Offset  | Data                                            | Ascii
[=] ----------------------------------------------------------------------------
[=]   0/0x00 | 01 48 4E 4C 31 01 00 00 10 10 00 03 18 AE 15 60 | .HNL1..........`
[=]  16/0x10 | 00 00 00 00 00 00 00 00 00 00 00 00 30 35 02 19 | ............05..
[=]  32/0x20 | 00 8D 39 DB C9 9B A3 7F 33 E3 22 89 97 BE CA EE | ..9.....3.".....
[=]  48/0x30 | AD C6 A5 9A EE C4 DC F0 60 02 18 68 CB A4 AE B1 | ........`..h....
[=]  64/0x40 | 6C 3F AA 9E 02 49 D9 DE 79 E1 13 6C D9 45 DF 10 | l?...I..y..l.E..
[=]  80/0x50 | 35 BB C7 00 00 00 00 00 00 00 00 00 00 00 00 00 | 5...............

[=] --------------------------------- File 01 ----------------------------------
[+] File ID         : 01
[+] File type       : Backup data  [0x01]
[+] Comm mode       : Full
[+] File size       : 48 [0x30] bytes
[+] Access rights   : ef10  (r: free w: disabled rw: key 0x01 change: key 0x00)
[!] File needs communication mode `encrypt` but there is no authentication
[=] Got file type: Backup data. Option: data. comm mode: encrypt
[=] ------------------------------- File 01 data -------------------------------
[+] Read 48 bytes from file 0x01 offset 0
[=]  Offset  | Data                                            | Ascii
[=] ----------------------------------------------------------------------------
[=]   0/0x00 | 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 | ................
[=]  16/0x10 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 | ................
[=]  32/0x20 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 | ................
```

There's an annotated dump of a different card with a transaction timestamp on the linked issue page as well.